session_db/pg_session_store.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) Odoo SA 2017
+# @author Nicolas Seinlet
+# Copyright (c) ACSONE SA 2022
+# @author Stéphane Bidoul
+import json
+import logging
+import os
+
+import psycopg2
+
+import odoo
+from odoo import http
+from odoo.tools._vendor import sessions
+from odoo.tools.func import lazy_property
+
+_logger = logging.getLogger(__name__)
+
+lock = None
+if odoo.evented:
+    import gevent.lock
+
+    lock = gevent.lock.RLock()
+elif odoo.tools.config["workers"] == 0:
+    import threading
+
+    lock = threading.RLock()
+
+
+def with_lock(func):
```

Comment:
This could possibly be replaced with `odoo.tools.func.locked`

Reply:
Interesting, does not seem to be a 1:1 replacement at first sight but could be interesting to investigate in another PR.